geos: also look at paths parenting os.Argv[0]

otan opened the pull request:
For cases like roachprod, the cockroach binary may be invoked from an
alternate directory. This may fail to find the appropriate GEOS library.
As such, also take into account os.Argv[0] when looking at parenting
directories.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/52380)
<!-- Reviewable:end -->

otan committed:
geos: also look at paths parenting os.Argv[0]

For cases like roachprod, the cockroach binary may be invoked from an
alternate directory. This may fail to find the appropriate GEOS library.
As such, also take into account os.Argv[0] when looking at parenting
directories.

Release note: None

ajwerner reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 2 of 2 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/52380#-:-MDyfSKK-7kMt2mjcP-o:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained



<!-- Sent from Reviewable.io -->

otan commented:
tftr!

bors r=ajwerner
